contrib/exporters/vpclogs/mod/transform.go

```diff
@@ -0,0 +1,186 @@
+	BytesFromInitiator             int64  `json:"bytes_from_initiator,omitempty"`
+	PacketsFromInitiator           int64  `json:"packets_from_initiator,omitempty"`
+	BytesFromTarget                int64  `json:"bytes_from_target,omitempty"`
+	PacketsFromTarget              int64  `json:"packets_from_target,omitempty"`
+}
+
+// VpclogsFlowTransformer is a custom transformer for flows
+type vpclogsFlowTransformer struct {
+	flowUpdateCountCache *cache.Cache
+	excludeStartedFlows  bool
+}
+
+func (ft *vpclogsFlowTransformer) setUpdateCount(f *flow.Flow) int64 {
+	var count int64
+	if countRaw, ok := ft.flowUpdateCountCache.Get(f.UUID); ok {
+		count = countRaw.(int64)
+	}
+
+	if f.FinishType != flow.FlowFinishType_TIMEOUT {
+		if f.FinishType == flow.FlowFinishType_NOT_FINISHED {
+			ft.flowUpdateCountCache.Set(f.UUID, count+1, cache.DefaultExpiration)
+		} else {
+			ft.flowUpdateCountCache.Set(f.UUID, count+1, time.Minute)
+		}
+	} else {
+		ft.flowUpdateCountCache.Delete(f.UUID)
+	}
+
+	return count
+}
+
+func (ft *vpclogsFlowTransformer) getStatus(f *flow.Flow, updateCount int64) string {
+	if f.FinishType != flow.FlowFinishType_NOT_FINISHED {
+		return "ENDED"
+	}
+
+	if updateCount == 0 {
+		return "STARTED"
+	}
+
+	return "UPDATED"
+}
+
+func (v *VpclogsFlow) SetAction(action string) {
+	v.Action = action
+}
+
+// DeriveExternalizedFields builds the fields that are visible in the vpc flow logs.
+// All these values are derived from values stored in the VpclogsFlow structure (not from the original flows) since these may have been updated by additional processing.
+func (v *VpclogsFlow) DeriveExternalizedFields() {
+	v.StartTime = v.LastUpdateMetric.Start
+	v.EndTime = v.LastUpdateMetric.Last
+	// In the meantime, EtherType is hard-coded
+	v.EtherType = "IPv4"
+	v.ConnectionStartTime = v.Start
+	v.CumulativeBytesFromInitiator = v.Metric.ABBytes
+	v.CumulativePacketsFromInitiator = v.Metric.ABPackets
+	v.CumulativeBytesFromTarget = v.Metric.BABytes
+	v.CumulativePacketsFromTarget = v.Metric.BAPackets
+	v.BytesFromInitiator = v.LastUpdateMetric.ABBytes
+	v.PacketsFromInitiator = v.LastUpdateMetric.ABPackets
+	v.BytesFromTarget = v.LastUpdateMetric.BABytes
+	v.PacketsFromTarget = v.LastUpdateMetric.BAPackets
+
+	// We don't usually see the report a finished flow
+	// TBM These 2 fields need to be handled differently
+	v.WasTerminated = v.FinishType != flow.FlowFinishType_NOT_FINISHED
+	v.WasInitiated = v.Status == "STARTED"
+
+	// IP protocol type (for example, TCP = 6; UDP = 17, ICMP = 1)
+	if v.Protocol == "TCP" {
+		v.TransportProtocol = 6
+	} else if v.Protocol == "UDP" {
+		v.TransportProtocol = 17
+	} else if v.Protocol == "ICMP" {
+		v.TransportProtocol = 1
+	}
+
+	// TBD fill in direction
+}
+
+// Transform transforms a flow before being stored
+func (ft *vpclogsFlowTransformer) Transform(f *flow.Flow) interface{} {
+	logging.GetLogger().Debugf("flow = %s", f)
```

Comment:
please remove debug messages left from development

Reply:
If logging level is not set to DEBUG, then these messages won't show up in the output. These messages (when enabled using the DEBUG flag) allow us to demonstrate to team-members (and to ourselves) that things are running as designed. There are numerous such Debug logging statements throughout the Skydive code.

Reply:
I suggest you minimize of such message only leaving the really essential ones